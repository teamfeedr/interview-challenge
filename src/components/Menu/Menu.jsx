import React from 'react';
import PropTypes from 'prop-types';

import MenuItem from '../MenuItem/MenuItem';

const getMenuItems = ({ items, handleSelectItem }) => {
    return items.map((item) => (
        <MenuItem
            key={item.name}
            id={item.id}
            name={item.name}
            dietaries={item.dietaries}
            handleOnSelect={() => {
                handleSelectItem({ item: { ...item } })
            }}
        />
    ));
}

const Menu = ({ items, handleSelectItem }) => {
    const menuItemElements = getMenuItems({ items, handleSelectItem });

    return (
        <ul className="item-picker">
            {menuItemElements}
        </ul>
    )
}

Menu.propTypes = {
    items: PropTypes.arrayOf(PropTypes.object).isRequired,
    handleSelectItem: PropTypes.func
}

Menu.defaultProps = {
    handleSelectItem: undefined
}

export default Menu