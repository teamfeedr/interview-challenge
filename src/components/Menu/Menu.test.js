import React from "react";
import { shallow } from "enzyme";

import Menu from './Menu';

describe("the Menu component", () => {
  it("renders with items", () => {
      const items = [
          {
              id: 1,
              name: 'test1',
              dietaries: ['v']
          },
          {
              id: 2,
              name: 'test2',
              dietaries: ['vg', 'gf']
          }
      ]

      const rendered = shallow(<Menu items={items}/>);
      expect(rendered).toMatchSnapshot();
  });
});
