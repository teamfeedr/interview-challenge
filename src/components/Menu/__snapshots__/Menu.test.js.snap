// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the Menu component renders with items 1`] = `
<ul
  className="item-picker"
>
  <MenuItem
    dietaries={
      Array [
        "v",
      ]
    }
    handleOnSelect={[Function]}
    id={1}
    key="test1"
    name="test1"
  />
  <MenuItem
    dietaries={
      Array [
        "vg",
        "gf",
      ]
    }
    handleOnSelect={[Function]}
    id={2}
    key="test2"
    name="test2"
  />
</ul>
`;
